tests/integration/RequestBuilder/EventsRequestBuilderTest.php

```diff
@@ -71,12 +75,39 @@ public function shouldExecuteInteractionAndUserMergeAndItemPropertyCommands(): v
         $this->assertResponseCommandStatuses($response, ...$this->generateOkStatuses(9));
     }
 
-    private static function addPropertiesToPropertySetupRequest(ItemPropertiesSetupRequestBuilder $builder): void
+    private static function setupItemProperties(Matej $matej): void
     {
-        $builder->addProperties([
-            ItemPropertySetup::string('test_property_a'),
-            ItemPropertySetup::string('test_property_b'),
-            ItemPropertySetup::string('test_property_c'),
-        ]);
+        $request = $matej->request()->setupItemProperties();
+        foreach (static::getPropertiesList() as $property) {
+            $request->addProperty(ItemPropertySetup::string($property));
+        }
+        $request->send();
+    }
+
+    private static function waitForItemPropertiesSetup(Matej $matej): void
+    {
+        while (true) {
+            $request = $matej->request()->getItemProperties();
+            $resp = $request->send();
+
+            $properties = [];
+            foreach ($resp->getData() as $property) {
+                $properties[] = $property->name;
+            }
+
+            if (!array_diff(static::getPropertiesList(), $properties)) {
+                return;
+            }
+            usleep(1000000); # 0.1s
```

Comment:
1000000 µs is 1 second, so it should be:

```suggestion
            usleep(100000); // 0.1s
```